### PR TITLE
Update user api doc.

### PR DESCRIPTION
**Pull request by ydye**

### CODE REVIEW

File: src/rest-server/docs/swagger.yaml [hunk 1]
```diff
@@ -278,38 +278,258 @@ paths:
...
+        - bearerAuth: []
       responses:
         '200':
           description: Succeeded
-          content: {}
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
```
Comment (mydmdm) [inline src/rest-server/docs/swagger.yaml:490]: this is the same to the item type in `get /api/v2/users`, right?

File: src/rest-server/docs/swagger.yaml [hunk 2]
```diff
@@ -330,19 +550,30 @@ paths:
...
     put:
       tags:
         - user
-        - group
       summary: Add a group for user.
-      description: Administrator add a group to other user's grouplist.
+      description: >-
+        Admin only.
+        Basic mode only.
+        Add a group to other user's grouplist.
```
Comment (mydmdm) [inline src/rest-server/docs/swagger.yaml:563]: only `other user`? An administrator could also add group to himself / herself?

File: src/rest-server/docs/swagger.yaml (hunk 1 above)
Comment (ydye) [reply src/rest-server/docs/swagger.yaml:490]: yes

File: src/rest-server/docs/swagger.yaml (hunk 2 above)
Comment (ydye) [reply src/rest-server/docs/swagger.yaml:563]: Yes, admin can add a group into himself/herslf.

File: src/rest-server/docs/swagger.yaml [hunk 3]
```diff
@@ -278,38 +278,258 @@ paths:
...
+                        When patch is true, it's optional.
+                    extension:
+                      type: object
+                      description: >-
+                        Extension fields.
+                        Be allowed in both oidc and basic mode.
+                        When patch is false, it's required.
+                        When patch is true, it's optional.
+                patch:
+                  type: Boolean
```
[the remark this answers is not in the document]
Comment (ydye) [reply src/rest-server/docs/swagger.yaml]: Fixed

File: src/rest-server/docs/swagger.yaml [hunk 4]
```diff
@@ -278,38 +278,258 @@ paths:
...
+                      description: Admin or not.
+                    virtualCluster:
+                      type: array
+                      description: The list of virtualCluster which the user can access.
+                      items:
+                        type: string
+                        description: virtual cluster name.
+                    storageConfig:
+                      type: object
+                      descrption: storageconfig
```
[the remark this answers is not in the document]
Comment (ydye) [reply src/rest-server/docs/swagger.yaml]: Fixed

File: src/rest-server/docs/swagger.yaml [hunk 5]
```diff
@@ -278,38 +278,258 @@ paths:
...
+                    description: Admin or not.
+                  virtualCluster:
+                    type: array
+                    description: The list of virtualCluster which the user can access.
+                    items:
+                      type: string
+                      description: virtual cluster name.
+                  storageConfig:
+                    type: object
+                    descrption: storageconfig
```
[the remark this answers is not in the document]
Comment (ydye) [reply src/rest-server/docs/swagger.yaml]: Fixed

**Pull request merged**